### DISCUSSION TITLE
Windows connect to specific Session

### DISCUSSION THREAD

**Discussion by HyP3r-** [rustdesk/rustdesk#2095] [Q&A] [answered]
We work a lot with Desktop Virtualization (VPS) and Remote Desktop Server (RDS). When our colleagues have problems, we have to connect to the correct session on Windows 10 (VPS) or Windows Server 2019 (RDS). For example, session 12 (as an example number). Now it would be just nice if before connecting RustDesk would ask "In which session do you want to connect?".

**Comment by HyP3r-** [comment]
Small addition to this. I had a look at the source code and had a closer look at the [windows.rs](https://github.com/rustdesk/rustdesk/blob/ee29db888af26e109fcebcab12f18ef80527cb55/src/platform/windows.rs) file. There is a function [get_current_session](https://github.com/rustdesk/rustdesk/blob/ee29db888af26e109fcebcab12f18ef80527cb55/src/windows.cc#L385). This function either returns the console session ID ([WTSGetActiveConsoleSessionId](https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-wtsgetactiveconsolesessionid)) or, if the registry key [share_rdp](https://github.com/rustdesk/rustdesk/blob/ee29db888af26e109fcebcab12f18ef80527cb55/src/platform/windows.rs#L699) is true, tries to return the session ID either from the console session or from the rdp session using [WTSEnumerateSessionsA](https://learn.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsenumeratesessionsa).

The goal of this discussion now would be to simply return WTSEnumerateSessionsA to the client and display a list of these sessions to the client. [The session ID that the client then selects will be used by RustDesk Server](https://github.com/rustdesk/rustdesk/blob/ee29db888af26e109fcebcab12f18ef80527cb55/src/platform/windows.rs#L596).

I hope this helps you with the implementation.

**Comment by rustdesk** [comment, marked as answer]
Ready in nightly build now, https://twitter.com/rustdesk/status/1759445426267189273
